app/services/foreman/provision/ssh.rb

```diff
@@ -72,7 +72,7 @@ def logger
   end
 
   def initiate_connection!
-    Timeout.timeout(360) do
+    Timeout.timeout(Setting[:ssh_timeout]) do
```

Comment:
I would add `.to_i`, that is: 
```ruby
Timeout.timeout(Setting[:ssh_timeout].to_i) do
``` 

in order to make sure that it is actually gets an integer rather then unknown type.

Reply:
Sure, I'll add that